sdks/go/pkg/beam/core/runtime/exec/fullvalue.go

```diff
@@ -151,6 +156,12 @@ func ConvertFn(from, to reflect.Type) func(interface{}) interface{} {
 			}
 			return ret.Interface()
 		}
+
+	case typex.IsList(from) && typex.IsUniversal(from.Elem()) && typex.IsUniversal(to):
+		return func(v interface{}) interface{} {
+			return Convert(v, to)
```

Comment:
Never call Convert from ConvertFn. The point of ConvertFn is to avoid re-computing the type comparisons, and the typex.IsUniversals and similar per element.  

Instead, write it the other way around. Convert can call ConvertFn, since it's already paying the heavier call cost anyway.

Reply:
Ar, I see. That's the reason that ConvertFn is cached by the callers.

So WDYT if we simply have Convert call ConvertFn for all the cases? (Or we should still avoid this indirection at least for some common cases like from == to?)

PTAL. Thanks!